### ISSUE TITLE
Splitting off popup-specific code into separate package?

### ISSUE THREAD

**Issue by wasamasa** [company-mode/company-mode#287]
I've seen auto-complete/popup-el#50 and am a bit sad about the proposed change not happening.  Would it be an option for `company-mode` to go ahead and split off its own popup into a separate library people could use who are interested in using a less buggy popup library?

**Comment by dgutov** [comment]
It's on my list. :)

`popup.el` has some extra features, though, like allowing several popups to be displayed at the same time, or supporting submenus. That would take some effort to replicate.

**Comment by moonlite** [comment]
Just a shot in the dark: but would it make sense to make company use popup.el instead? I use popup.el for yasnippet stuff and it would be great if company- and yasnippet popups looked the same. :)

**Comment by dgutov** [comment]
@moonlite Nope. You might want to read that other issue @wasamasa referenced.

**Comment by moonlite** [comment]
@dgutov Oh ok (sorry).

**Comment by dgutov** [comment]
@wasamasa By the way, have you tried `company-begin-with`?

**Comment by wasamasa** [comment]
@dgutov No, I didn't. Looks like a good alternative for the time being, although it would require me to use `company` itself as dependency and would probably not work if the user customized his copy to not use the popups.

**Comment by dgutov** [comment]
True.

**Comment by dgutov** [comment]
Well, actually, it would work, it just wouldn't use popups, which is probably what that user would want.

**Comment by dgutov** [comment]
I think [posframe](https://github.com/tumashu/posframe/) works pretty well now.

And since the upcoming Emacs 31 will support child frames in TTY too, that seems like a universal solution.

That doesn't include a menu-choice logic, though (does it need to?).
